module7-cloud-computing/r4-serverless-architecture/README.md

```diff
@@ -0,0 +1,106 @@
+
+## Why use serverless architectures?
+
+Hosting a software application on the internet usually involves managing some kind of server infrastructure. Typically this means a virtual or physical server that needs to be managed, as well as the operating system and other web server hosting processes required for your application to run. Using a virtual server from a cloud provider such as Amazon or Microsoft does mean the elimination of the physical hardware concerns, but still requires some level of management of the operating system and the web server software processes.
+
+With a serverless architecture, you focus purely on the individual functions in your application code. Services such as AWS Lambda and Microsoft Azure Functions take care of all the physical hardware, virtual machine operating system, and web server software management. You only need to worry about your code.
+
+## Advantages of serverless architecture
+
+serverless architecture offers the following value propositions:
+
+- Deploy & run. The infrastructure resources are managed by the cloud vendor. Internal IT can therefore focus on the business use case of software applications instead of managing the underlying hardware. Functions allow users to deploy application builds and configuration files necessary to provision the required hardware resources.
+- Fault tolerant. Since serverless application coding is logically decoupled from the underlying infrastructure, hardware failures have minimal impact on the software development process. Users are not required to manage applications on their own.
+- Low operational overhead. The infrastructure and operations management tasks are managed by cloud vendors, allowing organizations to focus their efforts on building software features. Applications are released faster, resulting in faster end-user feedback and therefore, continued improvements over the next software release cycles.
+- Optimized usage-based billing. The pay-as-you-go billing model serves particularly well for small and midsize (SMB) organizations that lack the capital to establish and manage on-site data centers.
+- Built-in integrations. Most cloud vendors offer integrations with a variety of services that allow users to focus on building high quality applications instead of configuring them.
+
+## Types of serverless architecture software
+
+There are three primary services offered via software developed with serverless architecture:
+
+### Function as a service
+
+In the realm of pre-packaged services, function as a service, sometimes known as FaaS or framework as, falls in between software as a service and platform as a service.
+
+Think of FaaS as a ready-to-implement framework that can be easily tailored to the needs of an enterprise company. To be clear:
+
+- SaaS is ready to use out of the box while FaaS is not.
+- However, FaaS does not require the resources to implement that you would need if you were using PaaS.
+
+### Backend as a service
```

Comment:
This paragraph is really hard to understand. This [article](https://www.cloudflare.com/learning/serverless/glossary/backend-as-a-service-baas/) covers it well. Some examples can be added like Firebase.